lib/dry/validation/evaluator.rb

```diff
@@ -155,16 +155,23 @@ def value
       #
       # This is useful when dealing with rules for optional keys
       #
-      # @example
+      # @example use the default key name
       #   rule(:age) do
       #     key.failure(:invalid) if key? && value < 18
       #   end
       #
+      # @example specify the key name
+      #   rule(:start_date, :end_date) do
+      #     if key?(:start_date) && !key?(:end_date)
+      #       key(:end_date).failure("must provide an end_date with start_date")
+      #     end
+      #   end
+      #
       # @return [Boolean]
       #
       # @api public
-      def key?
-        values.key?(key_name)
+      def key?(name = key_name)
```

Comment:
I think we want to support the same type of args as `Values#key?`, WDYT? Values supports symbols, arrays, path objects etc. so this could be something like:

```ruby
def key?(*args)
  args.empty? values.key?(key_name) : values.key?(*args)
end
```

Reply:
I was intending this to work with `Schema::Path`, yes.

The [`Values` implementation doensn't rely on splatting](https://github.com/dry-rb/dry-validation/blob/master/lib/dry/validation/values.rb#L64-L80) to work, is that necessary here?
```ruby
      def key?(key, hash = data)
        return hash.key?(key) if key.is_a?(Symbol)

        Schema::Path[key].reduce(hash) do |a, e|
          if e.is_a?(Array)
            result = e.all? { |k| key?(k, a) }
            return result
          elsif e.is_a?(Symbol) && a.is_a?(Array)
            return false
          else
            return false unless a.is_a?(Array) ? (e >= 0 && e < a.size) : a.key?(e)
          end
          a[e]
        end

        true
      end
```

Reply:
@alassek ah right, doesn't have to be `*args`, good point!